lib/activerecord-multi-tenant/relation_extension.rb

```diff
@@ -43,6 +43,8 @@ def generate_in_condition_subquery
       # Build an Arel query
       arel = if eager_loading?
                apply_join_dependency.arel
+             elsif ActiveRecord.gem_version >= Gem::Version.create('8.1.0')
+               arel()
```

Comment:
`arel()` memoizes `build_arel`.  

`update_all` uses this method, introduced in [Rails PR #54941](https://github.com/rails/rails/pull/54941).  
However, since `activerecord-multi-tenant` patches `build_arel` itself, it's unclear whether using `arel()` directly is entirely safe.

Reply:
Using `arel()` seems fine.

Since the implementation is nearly identical in 7.2, how about merging the 8.1 and 7.2 branches?
https://github.com/rails/rails/blob/v7.2.0/activerecord/lib/active_record/relation/query_methods.rb#L1572

```ruby
elsif ActiveRecord.gem_version >= Gem::Version.create('7.2.0')
  arel()
```

Reply:
Thanks!

context:
In Rails versions prior to 8.1, [update_all](https://github.com/rails/rails/blob/v7.2.0/activerecord/lib/active_record/relation.rb#L599) depends on `build_arel`. We will continue using `build_arel` for these versions to ensure consistency.